turbo.json

```diff
@@ -34,16 +34,19 @@
     },
     "dev": {
       "cache": false,
-      "dependsOn": ["build"],
+      "dependsOn": ["^compile"],
       "persistent": true
     },
```

Comment:
dev requires compile of dependencies, and vite will dynamically build all the typescript, so we don't need to depend on build. depending on build only adds a startup delay and demands a perfect build, instead of allowing warnings like dev. so this change allows faster development.

full build requirements are still present in all-check and are enforced by ci, so we lose nothing.